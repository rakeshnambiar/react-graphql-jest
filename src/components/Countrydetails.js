import React, {Component} from 'react';

import gpl from 'graphql-tag';
import {Query} from 'react-apollo';
import {graphql} from "graphql";


const COUNTRY_QUERY = gpl`
    query countryQuery($code: String!) {
        country(code: $code) {
            name,
            native,
            emoji,
            currency
        }
    }    
`;

class Countrydetails extends Component {
    constructor(props) {
        super(props);
        this.state = {
            inputValue: 'US',
            displayValue: 'US'
        };
        /*this.updateInputValue= this.updateInputValue.bind(this);*/
    }

    render() {
        return (

            <div>
                <h1>Country Details</h1>
                <input ref="countryCode" value={this.state.inputValue} onChange={evt => this.updateInputValue(evt)}/>
                <br/>
                {<Query query={COUNTRY_QUERY} variables={{"code": this.state.inputValue}}>
                    {
                        ({loading, error, data}) => {
                            if (loading) return <h4>Loading....</h4>;
                            if (error) return <h4>Status: Failed console.log(error)</h4>;
                            console.log(data);
                            //console.log(data.country);
                            let ct_name = '';
                            ct_name = data.country ? data.country.name : '';

                            //return ct_name;
                            return <div><textarea value={ct_name} onChange={this.onChange}/></div>

                        }
                    }
                </Query>}
                {/*<button onClick={() => this.displayValue()}>
                    Get Details
                </button>
                <br/>*/}


{/*                <textarea
                    id="countryDetails"
                    onChange={e => this.onChange(e)}
                    value={this.state.displayValue}
                />*/}
            </div>
        );
    }

    updateInputValue(evt) {
        this.setState({
            inputValue: evt.target.value,
        });
    }

    async displayValue() {
        this.setState({
            displayValue: this.refs.countryCode.value,
        });
    }

}

export default Countrydetails;
