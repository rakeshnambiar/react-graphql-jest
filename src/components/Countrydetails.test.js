import React from "react";
import {shallow} from 'enzyme';
import 'babel-polyfill';
import Countrydetails from './Countrydetails';


describe('Simple Math function', () => {
    test('adds 1 + 2 to equal 3', () => {
        expect(3).toBe(3);
    });
});

describe('Country Detail UI check', () => {
   test('Rendering the component', () => {
       const countryPage = shallow(
           <Countrydetails />
       );
       expect(countryPage).toMatchSnapshot();
   });
});

describe('Verify the input field', () => {
    it('Field exists or not', () => {
        const wrapper = shallow(<input/>);
        expect(wrapper.contains(<input/>)).toEqual(true);
    });
});


/*Posible Tests front-end

-- Fields are exists on not
-- Default value
-- Style for each ocmponent
-- Overall UI
-- Value changes
-- Nagative cases*/


/*
Posible Back-end Tests (Graphql)
--Field validation
--Insert is working currently
--Update is working currently
--Delete if any
--If any field removed
--If any field renamed
--In case any new field added*/
