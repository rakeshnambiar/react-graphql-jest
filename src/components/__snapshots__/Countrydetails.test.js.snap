// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Country Detail UI check Rendering the component 1`] = `ShallowWrapper {}`;
